lib/tapioca/compilers/dsl/active_record_typed_store.rb

```diff
@@ -0,0 +1,171 @@
+      #
+      #   sig { returns(T.nilable(Date)) }
+      #   def review_date; end
+      #
+      #   sig { returns(T.nilable(Date)) }
+      #   def review_date_was; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def review_date_changed?; end
+      #
+      #   sig { returns(T.nilable(Date)) }
+      #   def review_date_before_last_save; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def saved_change_to_review_date?; end
+      #
+      #   sig { returns(T.nilable([T.nilable(Date), T.nilable(Date)])) }
+      #   def review_date_change; end
+      #
+      #   sig { returns(T.nilable([T.nilable(Date), T.nilable(Date)])) }
+      #   def saved_change_to_review_date; end
+      #
+      #   sig { params(reviewd: T::Boolean).returns(T::Boolean) }
+      #   def reviewed=(reviewed); end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def reviewed; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def reviewed_was; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def reviewed_changed?; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def reviewed_before_last_save; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def saved_change_to_reviewed?; end
+      #
+      #   sig { returns(T.nilable([T::Boolean, T::Boolean])) }
+      #   def reviewed_change; end
+      #
+      #   sig { returns(T.nilable([T::Boolean, T::Boolean])) }
+      #   def saved_change_to_reviewed; end
+      # end
+      # ~~~
+      # end
+
+      class ActiveRecordTypedStore < Base
+        extend T::Sig
+
+        sig do
+          override
+            .params(
+              root: Parlour::RbiGenerator::Namespace,
+              constant: T.class_of(::ActiveRecord::Base)
+            )
+            .void
+        end
+        def decorate(root, constant)
+          stores = constant.typed_stores
+          return if stores.values.flat_map(&:accessors).empty?
+
+          root.path(constant) do |k|
+            stores.values.each do |store_data|
+              store_data.accessors.each do |accessor|
+                field = store_data.fields[accessor]
+                type = type_for(field.type_sym)
+                type = "T.nilable(#{type})" if field.null
+                generate_methods(field.name.to_s, type, k)
+              end
+            end
+          end
+        end
+
+        sig { override.returns(T::Enumerable[Module]) }
+        def gather_constants
+          ::ActiveRecord::Base.descendants.select do |klass|
+            klass.include?(ActiveRecord::TypedStore::Behavior)
+          end
+        end
+
+        private
+
+        TYPES =
+          {
+            boolean: "T::Boolean",
+            integer: "Integer",
+            string: "String",
+            float: "Float",
+            date: "Date",
+            time: "Time",
+            datetime: "DateTime",
+            decimal: "BigDecimal",
+            any: "T.untyped",
+          }
```

Comment:
Can we please have a test that has an example of each of these types? You can have multiple expectations that match to a single method and its signature for each of these. It would ensure that we are mapping everything to the correct type.